ash-bergs/plan-and-plate
# feat: recipe form functionality

Adds recipes to the database! 

- Adds `addRecipeFormDataAtom` - in this atom we initialize the form data values
- Adds `useFormData` hook to use our new atom - there are three inner functions returned from the hook to allow us to easily update a generic field, and for handling the more complex addition of steps and ingredients. 
- Updates /add page, using `useFormData` hook to give values and functionality to the form and form inputs 

To test:
1. `yarn dev` and navigate to the [Writer page](http://localhost:3000/recipes)
2. Fill out form and click save 
3. Return to the recipes page 
4. New recipe should be in the list
